### ISSUE TITLE
ConstProp references invalid subgraph for some reason

### ISSUE THREAD

**Issue by edwardcwang** [freechipsproject/firrtl#644]
Not sure what causes it, but the new ConstProp causes a crash in the following circuit.

Testcase with circuit below:

```scala
import firrtl._
import firrtl.passes._
import firrtl.transforms._

val input = """
circuit T_1090_ext :
  module T_1090_ext :
    input RW0_clk : Clock
    input RW0_addr : UInt<9>
    input RW0_wdata : UInt<64>
    output RW0_rdata : UInt<64>
    input RW0_en : UInt<1>
    input RW0_wmode : UInt<1>

    inst mem_0_0 of SRAM1RW512x32
    inst mem_0_1 of SRAM1RW512x32
    mem_0_0.clk <= RW0_clk
    mem_0_0.addr <= RW0_addr
    node RW0_rdata_0_0 = bits(mem_0_0.dout, 31, 0)
    mem_0_0.din <= bits(RW0_wdata, 31, 0)
    mem_0_0.write_en <= and(and(RW0_wmode, UInt<1>("h1")), UInt<1>("h1"))
    mem_0_1.clk <= RW0_clk
    mem_0_1.addr <= RW0_addr
    node RW0_rdata_0_1 = bits(mem_0_1.dout, 31, 0)
    mem_0_1.din <= bits(RW0_wdata, 63, 32)
    mem_0_1.write_en <= and(and(RW0_wmode, UInt<1>("h1")), UInt<1>("h1"))
    node RW0_rdata_0 = cat(RW0_rdata_0_1, RW0_rdata_0_0)
    RW0_rdata <= mux(UInt<1>("h1"), RW0_rdata_0, UInt<1>("h0"))

  extmodule SRAM1RW512x32 :
    input clk : Clock
    input addr : UInt<9>
    input din : UInt<32>
    output dout : UInt<32>
    input write_en : UInt<1>

    defname = SRAM1RW512x32

  module T_406_ext :
    input RW0_clk : Clock
    input RW0_addr : UInt<9>
    input RW0_wdata : UInt<64>
    output RW0_rdata : UInt<64>
    input RW0_en : UInt<1>
    input RW0_wmode : UInt<1>
    input RW0_wmask : UInt<8>

    inst mem_0_0 of SRAM1RW512x8
    inst mem_0_1 of SRAM1RW512x8
    inst mem_0_2 of SRAM1RW512x8
    inst mem_0_3 of SRAM1RW512x8
    inst mem_0_4 of SRAM1RW512x8
    inst mem_0_5 of SRAM1RW512x8
    inst mem_0_6 of SRAM1RW512x8
    inst mem_0_7 of SRAM1RW512x8
    mem_0_0.clk <= RW0_clk
    mem_0_0.addr <= RW0_addr
    node RW0_rdata_0_0 = bits(mem_0_0.dout, 7, 0)
    mem_0_0.din <= bits(RW0_wdata, 7, 0)
    mem_0_0.write_en <= and(and(RW0_wmode, bits(RW0_wmask, 0, 0)), UInt<1>("h1"))
    mem_0_1.clk <= RW0_clk
    mem_0_1.addr <= RW0_addr
    node RW0_rdata_0_1 = bits(mem_0_1.dout, 7, 0)
    mem_0_1.din <= bits(RW0_wdata, 15, 8)
    mem_0_1.write_en <= and(and(RW0_wmode, bits(RW0_wmask, 1, 1)), UInt<1>("h1"))
    mem_0_2.clk <= RW0_clk
    mem_0_2.addr <= RW0_addr
    node RW0_rdata_0_2 = bits(mem_0_2.dout, 7, 0)
    mem_0_2.din <= bits(RW0_wdata, 23, 16)
    mem_0_2.write_en <= and(and(RW0_wmode, bits(RW0_wmask, 2, 2)), UInt<1>("h1"))
    mem_0_3.clk <= RW0_clk
    mem_0_3.addr <= RW0_addr
    node RW0_rdata_0_3 = bits(mem_0_3.dout, 7, 0)
    mem_0_3.din <= bits(RW0_wdata, 31, 24)
    mem_0_3.write_en <= and(and(RW0_wmode, bits(RW0_wmask, 3, 3)), UInt<1>("h1"))
    mem_0_4.clk <= RW0_clk
    mem_0_4.addr <= RW0_addr
    node RW0_rdata_0_4 = bits(mem_0_4.dout, 7, 0)
    mem_0_4.din <= bits(RW0_wdata, 39, 32)
    mem_0_4.write_en <= and(and(RW0_wmode, bits(RW0_wmask, 4, 4)), UInt<1>("h1"))
    mem_0_5.clk <= RW0_clk
    mem_0_5.addr <= RW0_addr
    node RW0_rdata_0_5 = bits(mem_0_5.dout, 7, 0)
    mem_0_5.din <= bits(RW0_wdata, 47, 40)
    mem_0_5.write_en <= and(and(RW0_wmode, bits(RW0_wmask, 5, 5)), UInt<1>("h1"))
    mem_0_6.clk <= RW0_clk
    mem_0_6.addr <= RW0_addr
    node RW0_rdata_0_6 = bits(mem_0_6.dout, 7, 0)
    mem_0_6.din <= bits(RW0_wdata, 55, 48)
    mem_0_6.write_en <= and(and(RW0_wmode, bits(RW0_wmask, 6, 6)), UInt<1>("h1"))
    mem_0_7.clk <= RW0_clk
    mem_0_7.addr <= RW0_addr
    node RW0_rdata_0_7 = bits(mem_0_7.dout, 7, 0)
    mem_0_7.din <= bits(RW0_wdata, 63, 56)
    mem_0_7.write_en <= and(and(RW0_wmode, bits(RW0_wmask, 7, 7)), UInt<1>("h1"))
    node RW0_rdata_0 = cat(RW0_rdata_0_7, cat(RW0_rdata_0_6, cat(RW0_rdata_0_5, cat(RW0_rdata_0_4, cat(RW0_rdata_0_3, cat(RW0_rdata_0_2, cat(RW0_rdata_0_1, RW0_rdata_0_0)))))))
    RW0_rdata <= mux(UInt<1>("h1"), RW0_rdata_0, UInt<1>("h0"))

  extmodule SRAM1RW512x8 :
    input clk : Clock
    input addr : UInt<9>
    input din : UInt<8>
    output dout : UInt<8>
    input write_en : UInt<1>

    defname = SRAM1RW512x8

  module T_2172_ext :
    input W0_clk : Clock
    input W0_addr : UInt<6>
    input W0_data : UInt<88>
    input W0_en : UInt<1>
    input W0_mask : UInt<4>
    input R0_clk : Clock
    input R0_addr : UInt<6>
    output R0_data : UInt<88>
    input R0_en : UInt<1>

    inst mem_0_0 of SRAM2RW64x32
    inst mem_0_1 of SRAM2RW64x32
    inst mem_0_2 of SRAM2RW64x32
    inst mem_0_3 of SRAM2RW64x32
    mem_0_0.portB_clk <= W0_clk
    mem_0_0.portB_addr <= W0_addr
    mem_0_0.portB_din <= bits(W0_data, 21, 0)
    mem_0_0.portB_write_en <= and(and(UInt<1>("h1"), bits(W0_mask, 0, 0)), UInt<1>("h1"))
    mem_0_1.portB_clk <= W0_clk
    mem_0_1.portB_addr <= W0_addr
    mem_0_1.portB_din <= bits(W0_data, 43, 22)
    mem_0_1.portB_write_en <= and(and(UInt<1>("h1"), bits(W0_mask, 1, 1)), UInt<1>("h1"))
    mem_0_2.portB_clk <= W0_clk
    mem_0_2.portB_addr <= W0_addr
    mem_0_2.portB_din <= bits(W0_data, 65, 44)
    mem_0_2.portB_write_en <= and(and(UInt<1>("h1"), bits(W0_mask, 2, 2)), UInt<1>("h1"))
    mem_0_3.portB_clk <= W0_clk
    mem_0_3.portB_addr <= W0_addr
    mem_0_3.portB_din <= bits(W0_data, 87, 66)
    mem_0_3.portB_write_en <= and(and(UInt<1>("h1"), bits(W0_mask, 3, 3)), UInt<1>("h1"))
    mem_0_0.portA_clk <= R0_clk
    mem_0_0.portA_addr <= R0_addr
    node R0_data_0_0 = bits(mem_0_0.portA_dout, 21, 0)
    mem_0_1.portA_clk <= R0_clk
    mem_0_1.portA_addr <= R0_addr
    node R0_data_0_1 = bits(mem_0_1.portA_dout, 21, 0)
    mem_0_2.portA_clk <= R0_clk
    mem_0_2.portA_addr <= R0_addr
    node R0_data_0_2 = bits(mem_0_2.portA_dout, 21, 0)
    mem_0_3.portA_clk <= R0_clk
    mem_0_3.portA_addr <= R0_addr
    node R0_data_0_3 = bits(mem_0_3.portA_dout, 21, 0)
    node R0_data_0 = cat(R0_data_0_3, cat(R0_data_0_2, cat(R0_data_0_1, R0_data_0_0)))
    R0_data <= mux(UInt<1>("h1"), R0_data_0, UInt<1>("h0"))

  extmodule SRAM2RW64x32 :
    input portA_clk : Clock
    input portA_addr : UInt<6>
    output portA_dout : UInt<32>
    input portB_clk : Clock
    input portB_addr : UInt<6>
    input portB_din : UInt<32>
    input portB_write_en : UInt<1>

    defname = SRAM2RW64x32
"""
val state = CircuitState(firrtl.Parser.parse(input), UnknownForm)

val transforms = Seq(
      ToWorkingIR,
      ResolveKinds,
      InferTypes,
      ResolveGenders,
      InferWidths,
      new ConstantPropagation)

val finalState = transforms.foldLeft(state) {
  (c: CircuitState, t: Transform) => t.runTransform(c)
}

println(finalState.circuit.serialize)
```

Stack trace:
```
java.lang.IllegalArgumentException: requirement failed
  at scala.Predef$.require(Predef.scala:212)
  at firrtl.graph.DiGraph.subgraph(DiGraph.scala:314)
  at firrtl.transforms.ConstantPropagation.iterate$1(ConstantPropagation.scala:427)
  at firrtl.transforms.ConstantPropagation.run(ConstantPropagation.scala:466)
  at firrtl.transforms.ConstantPropagation.execute(ConstantPropagation.scala:485)
  at firrtl.Transform$$anonfun$4.apply(Compiler.scala:201)
  at firrtl.Transform$$anonfun$4.apply(Compiler.scala:201)
  at firrtl.Utils$.time(Utils.scala:135)
  at firrtl.Transform.runTransform(Compiler.scala:201)
  at $anonfun$1.apply(<console>:23)
  at $anonfun$1.apply(<console>:23)
  at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
  at scala.collection.immutable.List.foldLeft(List.scala:84)
  ... 44 elided
```

**Comment by edwardcwang** [comment]
The bug is caused by 672162b4bf6ca4a4a4ed7a4a9ffaadfea428ede0 (#633) - previous revisions are free of the bug.

**Comment by edwardcwang** [comment]
Attempted fix in #645
